File: rup-projects-core/src/main/java/com/core/rupprojectscore/dto/MemberDto.java
package com.core.rupprojectscore.dto;

import com.core.rupprojectscore.entity.Member;
import com.core.rupprojectscore.entity.MemberRole;
import com.core.rupprojectscore.entity.UseCase;
import com.core.rupprojectscore.service.Mapper;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MemberDto {
    private Long id;
    private String name;
    private String surName;
    private String email;
    private MemberRole role;

    public static MemberDto modelToDto(Member member) {
        return new Mapper().map(member, MemberDto.class);
    }

    public static Member dtoToModel(MemberDto dto) {
        return new Mapper().map(dto, Member.class);
    }

}

File: web-app/src/app/shared/services/dtos/member-dto.ts
import {Id} from '../../types/id';
import {ActivityDto} from './activity-dto';
import {MemberRole} from '../../types/member-role';

export interface MemberDto {

  id: Id;
  name: string;
  surName: string;
  email: string;
  role: MemberRole;
  activities: ActivityDto[];

}
